agents/appmotion_motor/appmotion_motors_agent.py

```diff
@@ -0,0 +1,1142 @@
+
+        self.agent.register_feed('positions',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    def init_motors_task(self, session, params=None):
+        """init_motors_task(params)
+
+        **Task** - Connect to the motors, either one or both.
+
+        Parameters:
+            params (dict): Parameters dictionary for passing
+                parameters to task.
+        """
+
+        if params is None:
+            params = {}
+        self.log.debug("Trying to acquire lock")
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn(
+                    "Could not start init because {} is already running".format(
+                        self.lock.job))
+                return False, "Could not acquire lock."
+
+            self.log.debug("Lock Acquired Connecting to Stages")
+
+            print('establishing serial server with motor1!')
+            self.motor1 = Motor(self.motor1_ip, self.motor1_port, self.motor1_is_lin, mot_id='motor1', index=1, m_res=self.m_res)
+            print('establishing serial server with motor2!')
+            self.motor2 = Motor(self.motor2_ip, self.motor2_port, self.motor2_is_lin, mot_id='motor2', index=2, m_res=self.m_res)
+
+        # This part is for the record and to allow future calls to proceed,
+        # so does not require the lock
+        self.initialized = True
+        if self.auto_acq:
+            self.agent.start('acq')
+        return True, 'Motor(s) Initialized.'
+
+    @ocs_agent.param('lin_stage', default=True, type=bool)
+    @ocs_agent.param('motor', default=1, type=int)
+    @ocs_agent.param('pos_is_inches', default=False, type=bool)
+    @ocs_agent.param('pos', default=0, type=float)
+    @ocs_agent.param('verbose', default=True, type=bool)
+    def move_axis_to_position(self, session, params):
+        """move_axis_to_position(motor=1, pos=0, pos_is_inches=False,
+            lin_stage=True)
+
+        ** Task** - Move the axis to the given absolute position in counts or
+        inches.
+
+        .. note::
+            If moving multiple axes, function will assume ``lin_stage`` value
+            for all axes.
+
+        Parameters:
+            motor (int): Determines which motor, either 1 or 2, 3 is for all
+                motors. (default 1)
+            pos (float): The desired position in counts or in inches, positive
+                indicates away from the motor (default 0)
+            pos_is_inches (bool): True if pos was specified in inches, False
+                if in counts (default False)
+            lin_stage (bool): True if the specified motor is for the linear
+                stage, False if not (default True)
+        """
+
+        self.move_status = self.is_moving(session, params)[1][1]
```

Comment:
Ah, some confusion:
We moved `self.is_moving()` _outside_ the with block because it has its own "lock context". Putting it inside the with block will yield an error like "can't acquire lock".